### PR TITLE
addressing MSJ feedback from further WGLC review

### PR DESCRIPTION
**Pull request by wessels**

### CODE REVIEW

File: draft-dns-zone-digest.xml [hunk 1]
```diff
@@ -402,7 +405,7 @@ An alternate method (rfc include) is described in the references. --><!ENTITY RF
         </section>
         <section title="General Purpose Comparison Check">
           <t>
-            Since the zone digest does not depend on presentation
+            Since the zone digest calculation does not depend on presentation
```
Comment (wkumari) [inline draft-dns-zone-digest.xml:408]: This is helpful.

File: draft-dns-zone-digest.xml [hunk 2]
```diff
@@ -444,8 +447,12 @@ An alternate method (rfc include) is described in the references. --><!ENTITY RF
...
           have no meaning in this specification.
           Non-apex &RRNAME; RRs MUST NOT be used for verification.
-          Non-apex &RRNAME; RRSets are treated like any other RRSet
-          (which is to say they are included) during digest calculation.
+        </t>
+        <t>
+          During digest calculation,
+          non-apex &RRNAME; RRs are treated like any other RRs.
+          They are digested as-is and the RR is not replaced
+          by a placeholder RR.
```
Comment (wkumari) [inline draft-dns-zone-digest.xml:455]: LGTM.

File: draft-dns-zone-digest.xml [hunk 3]
```diff
@@ -575,11 +582,14 @@ example.com. 86400 IN &RRNAME; 2018031500 1 1 (
...
           Prior to calculation of the digest, and prior to signing with
-          DNSSEC, a placeholder &RRNAME; record is added to the
+          DNSSEC, one or more placeholder &RRNAME; records are added to the
           zone apex.  This serves two purposes: (1) it allows the digest to cover
           the Serial, Scheme, and Hash Algorithm fields, and (2) ensures that 
           appropriate denial-of-existence (NSEC, NSEC3) records are created
           if the zone is signed with DNSSEC.
+          When multiple &RRNAME; RRs are published in the zone, e.g.,
+          during an algorithm rollover, each must specify a unique Scheme
+          and Hash Algorithm tuple.
```
Comment (wkumari) [inline draft-dns-zone-digest.xml:592]: LGTM - the original location also seemed find, but...

File: draft-dns-zone-digest.xml [hunk 4]
```diff
@@ -724,7 +729,10 @@ RR(i) = owner | type | class | TTL | RDATA length | RDATA
     <section title="Verifying Zone Digest" anchor="verifying">
       <t>
         The recipient of a zone that has a &RRNAME; RR can verify the zone
-        by calculating the digest as follows:
+        by calculating the digest as follows.
+          If multiple &RRNAME; RRs are present in the zone, e.g.,
+          during an algorithm rollover, a match using any one of the recipient's
+          supported Schemes and Hash Algorithms is sufficient to verify the zone.
```
Comment (wkumari) [inline draft-dns-zone-digest.xml:735]: LGTM.

File: draft-dns-zone-digest.xml [hunk 5]
```diff
@@ -1037,6 +1043,14 @@ RR(i) = owner | type | class | TTL | RDATA length | RDATA
           0.13 seconds to calculate a SIMPLE SHA384 digest for a zone with
           22,000 RRs, and about 2.5 seconds for a zone with 300,000 RRs.
         </t>
+        <t>
+          These benchmarks attempt to emulate a worst-case scenario and
+          take into account the time required to canonicalize the zone
+          for processing.
+          Each of the 800+ zones were measured three times, and then
+          averaged, with a different random sorting of the input data
+          prior to each measurement.
```
Comment (wkumari) [inline draft-dns-zone-digest.xml:1061]: LGTM.

File: draft-dns-zone-digest.xml [hunk 6]
```diff
@@ -839,6 +840,7 @@ RR(i) = owner | type | class | TTL | RDATA length | RDATA
     </section -->
 
     <section title="IANA Considerations" anchor="iana">
+      <!-- Whats requirement for adding new entries?  RFC or expert review? -->
```
Comment (wkumari) [inline draft-dns-zone-digest.xml]: I think it is Specification Required -- originally I was saying Expert Review, but with the scheme, one needs to make sure that implementers know how to sort / canonicalize the input.

File: draft-dns-zone-digest.xml [hunk 7]
```diff
@@ -953,14 +955,18 @@ RR(i) = owner | type | class | TTL | RDATA length | RDATA
...
+      <section title="Attacks Utilizing &RRNAME; Queries">
         <t>
           Nothing in this specification prevents clients from making,
-          and servers from responding to, &RRNAME; queries.  One might
-          consider how well &RRNAME; responses could be used in
-          a distributed denial-of-service amplification attack.
+          and servers from responding to, &RRNAME; queries.
+          Servers SHOULD NOT calculate zone digests dynamically (for
+          each query) as this can be used as a CPU resource exhaustion
+          attack.
```
Comment (wkumari) [inline draft-dns-zone-digest.xml:973]: Yeah, this is worth adding -- I didn't really think implementations might do this, but I guess I *could* see someone:
A: doing this on AXFR
B: allowing AXFR from arbritary places.

**Review by wkumari — APPROVED** [review]

**Comment by wessels** [comment]
merging.

**Pull request merged**